lib/src/main/java/io/cloudquery/helper/ArrowHelper.java

```diff
@@ -260,4 +259,23 @@ public static ByteString encode(Resource resource) throws IOException {
       }
     }
   }
+
+  public static Resource decodeResource(ByteString byteString)
+      throws IOException, ValidationException {
+    try (BufferAllocator bufferAllocator = new RootAllocator()) {
+      try (ArrowStreamReader reader =
+          new ArrowStreamReader(byteString.newInput(), bufferAllocator)) {
+        VectorSchemaRoot vectorSchemaRoot = reader.getVectorSchemaRoot();
+        reader.loadNextBatch();
```

Comment:
I think we would need to loop on `loadNextBatch` until it returns false, see https://arrow.apache.org/docs/java/reference/org/apache/arrow/vector/ipc/ArrowStreamReader.html#loadNextBatch--

Though maybe not an issue at the moment since we always send a single record batch?

Reply:
Let's keep it as is for now and change it if we run into issues